client/hotrod-client/src/main/java/org/infinispan/client/hotrod/marshall/ProtoStreamMarshaller.java

```diff
@@ -5,24 +5,17 @@
 import org.infinispan.commons.marshall.Marshaller;
 import org.infinispan.protostream.ProtobufUtil;
 import org.infinispan.protostream.SerializationContext;
-import org.infinispan.query.remote.client.BaseProtoStreamMarshaller;
```

Comment:
After this rename we have 2 ProtoStreamMarshaller. I feel a bit confused :)

Reply:
I agree this isn't ideal, but I don't think we can rename/remove the client `ProtoStreamMarshaller` for compatibility reasons.

We could call the marshaller in commons something else, but the only thing I can think of is the less descriptive `ProtoMarshaller`

Reply:
Ok. let's have them both named ProtoStreamMarshaller, but a line of javadoc should be added to org.infinispan.client.hotrod.marshall.ProtoStreamMarshaller to explain it exist for historical reasons. And the ```instanceof``` in ProtoStreamMarshaller.getSerializationContext(RemoteCacheManager remoteCacheManager) should refer to the one in commons.

Reply:
I have deprecated `org.infinispan.client.hotrod.marshall.ProtoStreamMarshaller` and converted all test use to the commons `ProtostreamMarshaller`. Furthermore, I have migrated internal use of `ProtostreamMarshaller::getSerializationContext` to `MarshallerUtil.getSerializationContext` as well as adding the convenience method `MarshallerRegistration::registerMarshallers(RemoteCacheManager)` to reduce verbosity.